authentication/models/users.py

```diff
@@ -0,0 +1,120 @@
+from django.contrib.auth.models import User
+from django.db import models
+
+from historical_surveillance.models import School, District
+from django.contrib.auth.models import User, Group
+from django.db.models.signals import post_save
+from django.dispatch import receiver
+
+
+class SchoolAdministrator(User):
+    school = models.ForeignKey(School, on_delete=models.CASCADE, null=True)
+
+    class Meta:
+        verbose_name = "School Administrator"
+
+
+class Teacher(User):
+    school = models.ForeignKey(School, on_delete=models.CASCADE, null=True)
+
+    class Meta:
+        verbose_name = "Teacher"
+
+
+class SchoolPrincipal(User):
+    school = models.ForeignKey(School, on_delete=models.CASCADE, null=True)
+
+    class Meta:
+        verbose_name = "Principal"
+
+
+class DistrictEducationOfficer(User):
+    district = models.ForeignKey(District, on_delete=models.CASCADE, null=True)
+
+    class Meta:
+        verbose_name = "District Education Officer"
+
+
+class SchoolSuperviser(User):
+    # TODO
+    class Meta:
+        verbose_name = "School Superviser"
+
+
+class StatisticianAdmin(User):
+    # TODO
+    class Meta:
+        verbose_name = "Stastician Admin"
+
+
+class EvaluationAdmin(User):
+    # TODO
+    class Meta:
+        verbose_name = "Evaluation Admin"
+
+
+class EarlyChildhoodEducator(User):
+    # TODO
+    class Meta:
+        verbose_name = "Early Childhood"
+
+
+class SupportServicesAdmin(User):
+    # TODO
+    class Meta:
+        verbose_name = "Support Services Admin"
+
+
+class ExternalAccessor(User):
+    # TODO
+    class Meta:
+        verbose_name = "External Accessor"
+
+
+"""
+All users that are able to login extend the User class
+When a new user is created they will be an instance of a class that inherits from User
+So whenever a User is saved the method below runs and
+Checks if a user is being created and then checks which subclass it is and assigns the correct permissions
+"""
+
+
+def add_user_permissions(sender, instance, created, **kwargs):
```

Comment:
Following the design on Pauls' PR: https://github.com/guezandy/PAS-EMIS/pull/10

Automatically when a User is created add it to the correct group based on its class.

Reply:
great